### PR TITLE
Set project_image to thumbnail when sharing Dance Party projects

### PR DESCRIPTION
**Pull request by Erin007**
### **Dance Party project social share before**

Currently when a Dance Party project is shared on Facebook, we use a placeholder image for the share. By going to the share page of a local project, inspecting the og:image tag and copying the url for the image, we can preview locally which image will be shared

dance-share-image-before

Which results in this post when a project is shared from production: 

screen shot 2018-11-20 at 9 47 38 am

### **Dance Party project social share after**

In this PR, I set the project_image for Dance Party projects to the thumbnail for the given project. 

By going to the share page of the local project, inspecting the og:image tag, we see that it is now customized to have the channel id for the given project and the path to  the thumbnail:

screen shot 2018-11-21 at 3 18 21 pm

And copying the url for the image, we can preview locally which image will be shared:

screen shot 2018-11-21 at 3 19 14 pm

### CODE REVIEW

**Comment by Erin007** [comment]
A few concerns that I'm still not 100% sure how to address: 

- This change is difficult to verify across environments. Before and after this change, sharing from staging and test does not format the Facebook share in a way that is consistent with shares from production. It looks like: 

screen shot 2018-11-21 at 3 27 27 pm

Similarly, the [Facebook Debugger](https://developers.facebook.com/tools/debug/sharing/?q=https%3A%2F%2Fstaging-studio.code.org%2Fprojects%2Fdance%2FCCacnCr33DZ04rDRUrVXwg) 
doesn't show the image for a shared staging project. 
[facebook-debugger.pdf](https://github.com/code-dot-org/code-dot-org/files/2606114/facebook-debugger.pdf)

- I'm fairly confident based on the explanation the description that this _will_ work on production, despite the difficulty verifying across environments. However, I have a remaining concern about the size of the image.  The thumbnails we store, and that I've accessed here, are smaller than the placeholder image we use, so I don't know how it will look in context.

**Comment by breville** [comment]
I'm not very confident about changing any social media tags until we see how they look on the relevant sites.  In the past we've experienced a variety of croppings etc. depending on what we post and how people are viewing it. 

It might be possible to build a fresh page (just using pegasus) with a sample image to see how that works.

@tanyaparker is our resident expert on this stuff.

**Comment by tanyaparker** [comment]
I tried to test this locally but got a few errors. Probably just something wrong with my local set up, but thought I'd mention. If someone else can pull the PR as part of reviewing?

Level 1:
image

Level 13:
image

```
> missing credentials, provide credentials with one of the following options:
  - :access_key_id and :secret_access_key
  - :credentials
  - :credentials_provider
```

**Comment by tanyaparker** [comment]
I created a test page code.org/test to and put a square image as the og tag. It won't be as nice and big as the current dance party og tag.

screen shot 2018-11-24 at 10 00 49 pm
screen shot 2018-11-24 at 10 01 07 pm

File: dashboard/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -324,12 +324,18 @@ def show
...
-      # point to those here
-      @project_video = "https://dance-api.code.org/videos/video-#{@view_options['channel']}.mp4"
-      @project_video_stream = dance_project_embed_video_projects_url(key: params[:key], channel_id: params[:channel_id])
-      replay_video_view_options unless sharing || readonly
+    end
+
+    if params[:key] == 'dance'
+      if DCDO.get('share_video_sharing_enabled', nil)
+        # TODO: elijah set up test subdomains for dance-api, and situationally
+        # point to those here
```
Comment (breville) [inline dashboard/app/controllers/projects_controller.rb:333]: Will this TODO be part of this PR or a different one?

Comment (Hamms) [reply dashboard/app/controllers/projects_controller.rb:333]: that's a different one

File: dashboard/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -324,12 +324,18 @@ def show
...
+        @project_video = "https://dance-api.code.org/videos/video-#{@view_options['channel']}.mp4"
+        @project_video_stream = dance_project_embed_video_projects_url(key: params[:key], channel_id: params[:channel_id])
+        replay_video_view_options unless sharing || readonly
+      else
+        @project_image = CDO.studio_url "v3/files/#{@view_options['channel']}/.metadata/thumbnail.png"
```
Comment (Hamms) [inline dashboard/app/controllers/projects_controller.rb]: We will definitely want `project_image` to be set independent of the share video sharing setting; this is almost certainly why we're seeing different behavior on production vs staging

**Comment by Erin007** [comment]
@Hamms Are these changes inline with what you were thinking?

**Pull request merged**